appeals/e2e/cypress/e2e/back-office-appeals/reviewLpaq.spec.js

```diff
@@ -2,67 +2,88 @@
 /// <reference types="cypress"/>
 
 import { users } from '../../fixtures/users';
-import { AppealsListPage } from '../../page_objects/appealsListPage';
-import { UpdateDueDatePage } from '../../page_objects/updateDueDatePage';
-const appealsListPage = new AppealsListPage();
-const updateDueDatePage = new UpdateDueDatePage();
+import { CaseDetailsPage } from '../../page_objects/caseDetailsPage.js';
+import { ListCasesPage } from '../../page_objects/listCasesPage';
+import { DateTimeSection } from '../../page_objects/dateTimeSection';
+import { urlPaths } from '../../support/urlPaths.js';
 
-describe('Appeals feature', () => {
+const listCasesPage = new ListCasesPage();
+const dateTimeSection = new DateTimeSection();
+const caseDetailsPage = new CaseDetailsPage();
+
+describe('Review LPAQ', () => {
 	beforeEach(() => {
 		cy.login(users.appeals.caseAdmin);
 	});
 
 	it('Complete LPAQ', () => {
-		cy.visit('/appeals-service/all-cases');
-		appealsListPage.clickAppealFromList(30);
-		appealsListPage.clickReviewLpaq(7);
-		appealsListPage.selectRadioButtonByValue('Complete');
-		appealsListPage.clickButtonByText('Continue');
-		appealsListPage.clickLinkByText('Go back to case details');
-		const status = 'Complete';
-		const testData = { rowIndex: 1, cellIndex: 0, textToMatch: status, strict: true };
-		appealsListPage.verifyTableCellText(testData);
+		cy.createCase().then((caseRef) => {
+			cy.addLpaqSubmissionToCase(caseRef).then(() => {
+				cy.visit(urlPaths.appealsList);
+				listCasesPage.clickAppealByRef(caseRef);
+				caseDetailsPage.clickReviewLpaq();
+				caseDetailsPage.selectRadioButtonByValue('Complete');
+				caseDetailsPage.clickButtonByText('Confirm');
+				caseDetailsPage.clickLinkByText('Go back to case details');
+
+				const status = 'Complete';
+				const testData = { rowIndex: 1, cellIndex: 0, textToMatch: status, strict: true };
+				listCasesPage.verifyTableCellText(testData);
+			});
+		});
 	});
 
 	it('incomplete LPAQ', () => {
-		cy.visit('/appeals-service/all-cases');
-		appealsListPage.clickAppealFromList(30);
-		appealsListPage.clickReviewLpaq(7);
-		appealsListPage.selectRadioButtonByValue('Incomplete');
-		appealsListPage.clickButtonByText('Continue');
-		appealsListPage.chooseCheckboxByIndex(1);
-		appealsListPage.fillInput1('Hello here is some extra info, have a nice day 7384_+!£ =');
-		appealsListPage.clickButtonByText('Continue');
-		updateDueDatePage.enterDateDay('19');
-		updateDueDatePage.enterDateMonth('12');
-		updateDueDatePage.enterDateYear('2024');
-		appealsListPage.clickButtonByText('Save and Continue');
-		appealsListPage.clickButtonByText('Confirm');
-		appealsListPage.clickLinkByText('Go back to case details');
-		const status = 'Incomplete';
-		const testData = { rowIndex: 1, cellIndex: 0, textToMatch: status, strict: true };
-		appealsListPage.verifyTableCellText(testData);
+		let futureDate = new Date();
+		futureDate.setMonth(futureDate.getDate() + 28);
```

Comment:
Is this meant to be `setDate`? Though I suppose the exact date isn't important!

Reply:
`setDate` would set the specific day number component within the date object, whereas in this case I'd like to add 28 days (4 weeks) to the current date. The thinking being that this will most likely pick a working day (Monday to Friday) which is required. However it cannot pick a bank holiday, so this test will sometimes fail.

I wanted to revisit this extra in general. In the past I've typically used a DateTime library in JavaScript, being the native handling is clunky. I'll post a message on the Core Services check-in to get peoples thoughts

Reply:
Update: I misunderstand the original intent of your comment. Yes, this should have been `futureDate.setDate` not `futureDate.setMonth`